compiler/optimizer/LocalAnalysis.cpp

```diff
@@ -714,7 +714,12 @@ bool TR_LocalAnalysisInfo::isCallLike(TR::Node *node) {
 
    if (node->getOpCode().hasSymbolReference())
       {
-      if (node->getSymbolReference()->getSymbol()->isVolatile() || node->getSymbolReference()->getSymbol()->isMethodMetaData())
+      if (node->getSymbolReference()->getSymbol()->isVolatile() || 
+          (node->getSymbolReference()->getSymbol()->isMethodMetaData()
+#if J9_PROJECT_SPECIFIC
```

Comment:
I think you could create a query in the front end that allows you to eliminate the if J9_PROJECT_SPECIFIC, something along the lines of isImmutableMetadataField might capture the category of interest here in LocalAnalysis.cpp. The J9 implementation of that would return true specifically for the offset you have allowed to be commoned.